pkg/controllers/state/suite_test.go

```diff
@@ -1263,6 +1263,15 @@ var _ = Describe("Cluster State Sync", func() {
 		}
 		Expect(cluster.Synced(ctx)).To(BeFalse())
 	})
```

Comment:
Worthwhile to add a test case to ensure that we only store a single state Node when we call UpdateNodeClaim directly without a provider id, call reconcile with a Nodeclaim without a providerid, call reconcile with the same NodeClaim that resolved its provider id, and then delete the Nodeclaim to ensure it all cleans-up appropriately?

Reply:
I don't think we can do exactly all of this, since we're only blocking Synced() on these nodeclaims. I'll add tests that ensure these nodeclaims are cleaned up on a deletion though.